### ISSUE TITLE
Cannot figure get essentials to make a save file

### ISSUE THREAD

**Issue by Goobstank** [Tyrsis/EssentialsPlugin#85]
I apologize for reposting this in here as I have posted in the KSH thread as well... 

Incredible work... 

I have been working on getting this tweaked for my server and so far its gone as easy as 123 except one weird thing.

My problem has been that I am not able to get the archived backups using Essentials to save to any folder that I specify or any folder at all for that matter. Im not sure im doing anything to effect this but the saves are just not in the appropriate designated folders or in any of the expected locations (Users//AppData/Roaming/SpaceEngineersDedicated/Saves). I do see that I am getting regular saves there that are working fine but no archived saves. WHERE ARE THE SAVES? In essentials the sub directories are enabled and the backups are enabled etc... is there some specific issue that could be causing this? I have looked in ProgramData too! I have the BackupCleanup Disabled and have specified a folder for the saves. I have manually looked in a number of places that may contain SE saves and no luck. Im sure when I find out what im doing wrong I'll kick myself but for now im pulling my hair out on this issue. Im sure Im missing something. 

Please help

**Comment by McyD** [comment]
You specify the backup location in the Backup setup area of the config, I am pretty sure the folder has to be empty as well, or it fails.  I use C:\SE backups, for example.  I am guessing that is your question, as you seem to be referring to saves meaning backups?

**Comment by Goobstank** [comment]
yes I double checked all the settings in essentials and i now have the folder set to C:\Saves and there is nothing in the folder ever.  I have tried about 6 different locations and restarted and reloaded the server and plugin on these occasions. I have the backups set for every 15 minutes.in Extender as well as in essentials. and it is autosaving.perfectly but there are no backups being put in this folder or any other folder I select at any time. This is also a totally clean install. 

BackupAsteroids                        False
BackupBaseDirectory                 C:\SAVES
BackupCleanupTime                  0
BackupCreateSubDirectories     True
BackupEnabled                          True
BackupEssentials                       False
BackupItems                               {set for 15 minute intervals} 

the settings in essentials look like this currently but I have tried several other settings and none of them worked. Is there some sort of write permission or error log that I can look at that would explain the issue? I have tried this on both a brand new 2008 R2 winserver as well as win7sp1 machine to no avail

**Comment by McyD** [comment]
You keep using the wrong terminology... Backups are not saves.  You say you entered a backup every fifteen minutes?  So you entered 96 backup items?

**Comment by Goobstank** [comment]
Thanks for responding please forgive my ignorance on the terminology. Yes you are correct I am attempting to make a backup every 15 minutes.. not sure where youre getting the number 96 from.

**Comment by McyD** [comment]
If you are trying to make a back up every 15 minutes, then you had to have entered 96 backup times.  They are set by time, not interval; as I showed above.  If you break the day into 15 minute increments, their are 96.  So unless you entered 96 individual back up times, you have set it wrong.

**Comment by Goobstank** [comment]
Ok that worked thank you immensely! :)

**Comment by Goobstank** [comment]
Ok its not working again... am I to understand that the plugin will not write to its own backup folder if there is already a backup in there? So if I want a 15 backup interval to archive every day all day, I have to have a script that will move the backups out of that folder after every save?

**Comment by McyD** [comment]
no, it will write to folders it has put files in, it just has to be empty initially.  I think you are probably breaking it, I doubt it was meant to be ran constantly like you are trying.

Most people use a 10-15 minutes save, and a back-up every 1-6 hours.  Really 15 minute backups are pointless IMHO, as that is the same as just force closing the server and rolling back to the last save...  I myself have only had to roll back a few times, about 4, having a back up within a few hours is plenty.

Basically, I suspect you are breaking something with trying to do so many backups constantly.

**Comment by Goobstank** [comment]
Hello again! 
How could I be breaking something? What could it be? I really wish I had any aptitude for coding. But I would like to make this work. One way or the other. The frequency of the backups is essential. I would also like to state that  before it would not back up even one time ever. I was able to get it to save one time after using your suggestion. I have set it up totally fresh with one save interval and ran it for about 48hrs and never got one save even in an empty folder. The only time I was able to get it to save was after I set up the multiple intervals. Regardless I need to have a script or instructions to use essentials to make 15 minute backups. I really appreciate your effort to help me. I cant possibly imagine what the issue is with this feature not working. I understand that it would be incredibly simple for me to manually save the sandbox files 2-6 times a day as needed but that is not my intention. I suppose I will have to either find a script or write one (not my strong point) that will perform the function I require. Thanks for your efforts and if you could help me further in anyway it would not go unnoticed. <3

**Comment by McyD** [comment]
How are you entering your backup items?  I am pretty sure that is what you are doing wrong.  You are still mixing up your terms I believe.

For example I have

Hour: 05
Minute: 30

Hour: 11
Minute: 30

Hour: 17
Minute: 30

Hour: 23
Minute: 30

This results in a total of four backups, one every 6 hours.

If you are doing a backup every fifteen minutes, you should have entered 96 of these...


A Save is separate from a backup.

An Auto-Save, overwrites/updates the save file.
A backup: copies the current save file to a different location for archiving.

**Comment by Goobstank** [comment]
yep there is no misunderstanding here I am doing this exactly :) its ok I had a friend write me a python script that just copies and compresses the save folder into a backup folder with a time and date. I just disabled the essentials save function as it apparently wont work for me on the machine im using

**Comment by Goobstank** [comment]
Now i tried to set up the Auto Restart and thats not working either. It will start the extender but not actually press start. ahh whoa is me lol,,, I should rename this thread to please help goob make this shit work thread.
